examples/multi-modal-data-movielens/02-Data-Enrichment.ipynb

```diff
@@ -0,0 +1,472 @@
+       "      <td>113041</td>\n",
+       "      <td>11862.0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   movieId  imdbId   tmdbId\n",
+       "0        1  114709    862.0\n",
+       "1        2  113497   8844.0\n",
+       "2        3  113228  15602.0\n",
+       "3        4  114885  31357.0\n",
+       "4        5  113041  11862.0"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "links.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "62423"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "links.imdbId.nunique()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tqdm import tqdm\n",
+    "import pickle"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "movies_infos = {}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "for cnt, row in tqdm(links.iterrows(), total=len(links)):\n",
+    "    movieID = row['movieId']\n",
+    "    imdbID = row['imdbId']\n",
+    "    \n",
+    "    if movieID in movies_infos:\n",
+    "        #print(\"Movie exist: %d\"%movieID)\n",
+    "        continue\n",
+    "    else:\n",
+    "        try:\n",
+    "            movie_info = ia.get_movie(imdbID)\n",
+    "            movies_infos[movieID] = movie_info\n",
+    "        except Exception as e:\n",
+    "            print(\"Movie %d download error\"%movieID, e)\n",
+    "\n",
+    "        if (cnt+1) %500 == 0:\n",
+    "            print(\"Saving to file, %d/%d\"%(cnt,len(links)))\n",
+    "            with open('movies_info.pkl', 'wb') as f:\n",
+    "                pickle.dump({\"movies_infos\": movies_infos}, f, protocol=pickle.HIGHEST_PROTOCOL)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('movies_info.pkl', 'wb') as f:\n",
+    "    pickle.dump({\"movies_infos\": movies_infos}, f, protocol=pickle.HIGHEST_PROTOCOL)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "len(movies_infos)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Scraping movie posters\n",
+    "\n",
+    "The movie metadata also contains link to poster images. We next collect these posters where available.\n",
+    "\n",
+    "Note: this process will take some time to complete."
```

Comment:
For this part, it doesn't look like there's a reasonable alternative source. Could this be sped up by parallelizing with `multiprocessing` or `joblib`?

Reply:
yep should be straightforward to parallelize, and the impact should be good given this is I/O bound

Reply:
I don't have time to immediately look into these improvements for now. Maybe we can table it for the next iteration :)

Reply:
Yeah, no worries, that works. I'm working on a prototype of retrieval->filtering->scoring->ordering with Triton and Tensorflow now. Once I have the pieces of that pulled together, would be great to see if we can incorporate multi-modal features based on your work in this PR.

Reply:
parallelizing now with multi workers